pipeline/signed-container-image-listener.groovy

```diff
@@ -0,0 +1,93 @@
+/*
+    Pipeline script for signed/rc compose listener
+*/
+// Global variables section
+def nodeName = "centos-7"
+def sharedLib
+def versions
+def cephVersion
+def composeUrl
+
+// Pipeline script entry point
+node(nodeName) {
+    timeout(unit: "MINUTES", time: 30) {
+        stage('Install prereq') {
+            if (env.WORKSPACE) {
+                deleteDir() }
+            checkout([
+                $class: 'GitSCM',
+                branches: [[name: 'refs/remotes/origin/rc-container-details']],
+                doGenerateSubmoduleConfigurations: false,
+                extensions: [[
+                    $class: 'SubmoduleOption',
+                    disableSubmodules: false,
+                    parentCredentials: false,
+                    recursiveSubmodules: true,
+                    reference: '',
+                    trackingSubmodules: false]],
+                submoduleCfg: [],
+                userRemoteConfigs: [[
+                    url: 'https://github.com/red-hat-storage/cephci.git']]
+            ])
+            sharedLib = load("${env.WORKSPACE}/pipeline/vars/lib.groovy")
+            sharedLib.prepareNode()
+        }
+    }
+
+    stage("Update Release File") {
+        def cimsg = sharedLib.getCIMessageMap()
+        versions = sharedLib.fetchMajorMinorOSVersion("signed-container-image")
+        def composePath = cimsg["compose-path"].replace("/mnt/redhat/", "")
+        composeUrl = "http://download-node-02.eng.bos.redhat.com/${composePath}".toString()
+        cephVersion = sharedLib.fetchCephVersion(composeUrl)
+        def location = "/ceph/cephci-jenkins/latest-rhceph-container-info"
+        def fileName = "RHCEPH-${versions.major_version}.${versions.minor_version}.yaml"
+        def fileExists = sh (returnStatus: true, script: "ls -l ${location}/${fileName}")
+        if (fileExists != 0) {
+            currentBuild.result = 'ABORTED'
+            error "File:${fileName} does not exist...." }
```

Comment:
This is checked in the `readFromReleaseFile`... can we avoid it

Reply:
If File does not exist "readFromReleaseFile" will return an empty map

Reply:
I am not sure whether its a valid scenario or not, but in case file exist and there no data in it (may be due to write failure from other listener) current approach is valid.

readFromReleaseFile - will not abort